### ISSUE TITLE
range-builtin-not-iterating false positive when used with secrets

### ISSUE THREAD

**Issue by nth10sd** [PyCQA/pylint#4760]
### Bug description

```python
$ cat test.py
"""testcase"""

import secrets

print(secrets.choice(range(0, 2)))
```

### Command used

```shell
pylint test.py --enable range-builtin-not-iterating
```

### Pylint output

```shell
************* Module test
test.py:5:21: W1638: range built-in referenced when not iterating (range-builtin-not-iterating)

------------------------------------------------------------------
Your code has been rated at 5.00/10 (previous run: 5.00/10, +0.00)
```

### Expected behavior

I'd presume that `range-builtin-not-iterating` is not triggered, because the `choice` function in `secrets` accept sequences, which is a type of iterator.

See #2731.

### Pylint version

```shell
$ pylint --version
pylint 2.9.5
astroid 2.6.5
Python 3.9.6 (default, Jul  3 2021, 16:40:50)
[GCC 9.3.0]
```

### OS / Environment

```
$ lsb_release -a
No LSB modules are available.
Distributor ID: Ubuntu
Description:    Ubuntu 20.04.2 LTS
Release:        20.04
Codename:       focal
```
(on WSL2)

**Comment by Pierre-Sassoulas** [comment]
Closing as won't do because ``range-builtin-not-iterating`` was removed from pylint.
